File: Example/javaapp/src/main/java/com/stripe/example/javaapp/fragment/location/HeaderViewHolder.java
package com.stripe.example.javaapp.fragment.location;

import android.content.res.Resources;
import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

import com.stripe.example.javaapp.R;
import com.stripe.example.javaapp.databinding.ListItemHeaderBinding;

/**
 * Bind Info about a location list to a header view.
 */
public class HeaderViewHolder extends RecyclerView.ViewHolder {
    private Resources resources;
    private ListItemHeaderBinding binding;

    public HeaderViewHolder(View parent) {
        super(parent);
        resources = parent.getResources();
        binding = ListItemHeaderBinding.bind(parent);
    }

    public void bind(int size, boolean hasMore) {
        int titleId = hasMore ? R.string.select_location_list_title_more : R.string.select_location_list_title;
        binding.listItemHeader.setText(resources.getString(titleId, size));
    }
}

File: Example/kotlinapp/src/main/java/com/stripe/example/fragment/location/HeaderViewHolder.kt
package com.stripe.example.fragment.location

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.stripe.example.R
import com.stripe.example.databinding.ListItemHeaderBinding

/**
 * Bind Info about a location list to a header view.
 */
class HeaderViewHolder(
    parent: View,
) : RecyclerView.ViewHolder(parent) {
    private val resources = parent.resources
    private val binding = ListItemHeaderBinding.bind(parent)

    fun bind(size: Int, hasMore: Boolean) {
        val titleId = if (hasMore) R.string.select_location_list_title_more else R.string.select_location_list_title
        binding.listItemHeader.text = resources.getString(titleId, size)
    }
}
